engine/gamesys/src/gamesys/resources/res_render_prototype.cpp

```diff
@@ -53,28 +60,106 @@ namespace dmGameSystem
+            render_resource_metas.SetCapacity(total_resource_count);
+            prototype->m_RenderResources.SetCapacity(total_resource_count);
+
             for (uint32_t i = 0; i < prototype_desc->m_Materials.m_Count; ++i)
             {
-                dmGameSystem::MaterialResource* material;
-                if (dmResource::RESULT_OK == dmResource::Get(factory, prototype_desc->m_Materials[i].m_Material, (void**)&material))
-                    prototype->m_Materials.Push(material);
-                else
+                void* material;
+                result = dmResource::Get(factory, prototype_desc->m_Materials[i].m_Material, (void**)&material);
+                if (result != dmResource::RESULT_OK)
                     break;
+
+                RenderResourceMeta meta;
+                meta.m_Name = prototype_desc->m_Materials[i].m_Name;
+                meta.m_Type = dmRender::RENDER_RESOURCE_TYPE_MATERIAL;
+
+                prototype->m_RenderResources.Push(material);
+                render_resource_metas.Push(meta);
             }
-            if (!prototype->m_Materials.Full())
+
+            if (prototype_desc->m_RenderResources.m_Count > 0)
             {
-                result = dmResource::RESULT_OUT_OF_RESOURCES;
+                // Supported render resource types:
+                dmResource::ResourceType res_type_material;
+                dmResource::ResourceType res_type_render_target;
+
+                dmResource::Result type_res = dmResource::GetTypeFromExtension(factory, "materialc", &res_type_material);
+                assert(type_res == dmResource::RESULT_OK);
+
+                type_res = dmResource::GetTypeFromExtension(factory, "render_targetc", &res_type_render_target);
+                assert(type_res == dmResource::RESULT_OK);
+
+                for (uint32_t i = 0; i < prototype_desc->m_RenderResources.m_Count; ++i)
+                {
+                    void* res;
+                    result = dmResource::Get(factory, prototype_desc->m_RenderResources[i].m_Path, &res);
+                    if (result != dmResource::RESULT_OK)
+                        break;
+
+                    dmResource::ResourceType res_type;
+                    dmResource::GetType(factory, res, &res_type);
+                    dmRender::RenderResourceType render_resource_type = dmRender::RENDER_RESOURCE_TYPE_INVALID;
+
+                    if (res_type == res_type_material)
+                    {
+                        render_resource_type = dmRender::RENDER_RESOURCE_TYPE_MATERIAL;
+                    }
+                    else if (res_type == res_type_render_target)
+                    {
+                        render_resource_type = dmRender::RENDER_RESOURCE_TYPE_RENDER_TARGET;
+                    }
+                    else
+                    {
+                        result = dmResource::RESULT_NOT_SUPPORTED;
+                        break;
+                    }
```

Comment:
Is it good enough to check the extension in the name instead of checking the actual type information?

Reply:
I think the lines got messed up here for some reason, I was referring to getting the type from the extension and comparing that to the fixed type pointers we get above instead of just comparing to the extension string